pallets/subtensor/src/swap.rs

```diff
@@ -396,4 +501,187 @@ impl<T: Config> Pallet<T> {
+            weight.saturating_accrue(T::DbWeight::get().reads_writes(2, 2));
+        }
+    }
+
+    /// Checks if a coldkey has any associated hotkeys.
+    ///
+    /// # Arguments
+    ///
+    /// * `coldkey` - The AccountId of the coldkey to check.
+    ///
+    /// # Returns
+    ///
+    /// * `bool` - True if the coldkey has any associated hotkeys, false otherwise.
+    pub fn coldkey_has_associated_hotkeys(coldkey: &T::AccountId) -> bool {
+        Owner::<T>::iter().any(|(_, owner)| owner == *coldkey)
+    }
+
+    /// Swaps the subnet owner from the old coldkey to the new coldkey for all networks where the old coldkey is the owner.
+    ///
+    /// # Arguments
+    ///
+    /// * `old_coldkey` - The AccountId of the old coldkey.
+    /// * `new_coldkey` - The AccountId of the new coldkey.
+    /// * `weight` - Mutable reference to the weight of the transaction.
+    ///
+    /// # Effects
+    ///
+    /// * Updates the subnet owner to the new coldkey for all networks where the old coldkey was the owner.
+    /// * Updates the transaction weight.
+    pub fn swap_subnet_owner_for_coldkey(
+        old_coldkey: &T::AccountId,
+        new_coldkey: &T::AccountId,
+        weight: &mut Weight,
+    ) {
+        for netuid in 0..=TotalNetworks::<T>::get() {
+            let subnet_owner = SubnetOwner::<T>::get(netuid);
+            if subnet_owner == *old_coldkey {
+                SubnetOwner::<T>::insert(netuid, new_coldkey.clone());
+                weight.saturating_accrue(T::DbWeight::get().writes(1));
+            }
+        }
+        weight.saturating_accrue(T::DbWeight::get().reads(TotalNetworks::<T>::get() as u64));
+    }
+
+    /// Swaps the owned hotkeys for the coldkey
+    ///
+    /// # Arguments
+    ///
+    /// * `old_coldkey` - The AccountId of the old coldkey.
+    /// * `new_coldkey` - The AccountId of the new coldkey.
+    /// * `weight` - Mutable reference to the weight of the transaction.
+    ///
+    /// # Effects
+    ///
+    /// * Updates the subnet owner to the new coldkey for all networks where the old coldkey was the owner.
+    /// * Updates the transaction weight.
+    pub fn swap_owned_for_coldkey(
+        old_coldkey: &T::AccountId,
+        new_coldkey: &T::AccountId,
+        weight: &mut Weight,
+    ) {
+        // Update OwnedHotkeys map with new coldkey
+        let hotkeys = OwnedHotkeys::<T>::get(old_coldkey);
+        OwnedHotkeys::<T>::remove(old_coldkey);
+        OwnedHotkeys::<T>::insert(new_coldkey, hotkeys);
+        weight.saturating_accrue(T::DbWeight::get().reads_writes(0, 2));
+    }
+
+    /// Returns the cost of swapping a coldkey.
+    ///
+    /// # Returns
+    ///
+    /// * `u64` - The cost of swapping a coldkey in Rao.
+    ///
+    /// # Note
+    ///
+    /// This function returns a hardcoded value. In a production environment, this should be configurable or determined dynamically.
+    pub fn get_coldkey_swap_cost() -> u64 {
```

Comment:
Todo?

Reply:
We actually need to make these free. Thanks !